### PR TITLE
[MonoVM] Improve thread state debugging and make it usable on Android

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: src/mono/mono/utils/mono-threads-api.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ typedef struct _MonoStackData {
 } MonoStackData;
 
 // FIXME an ifdef to change __func__ to empty or further minimization.
-#ifndef DISABLE_LOGGING
+#if !defined(DISABLE_LOGGING) || defined(DEBUG)
```
Comment (lambdageek) [inline src/mono/mono/utils/mono-threads-api.h:30]: what if we don't disable logging in debug builds, instead? 

https://github.com/dotnet/runtime/blob/74b18d3c5e910ba2d075f075eede0543b03983fb/src/mono/mono.proj#L655

Comment (filipnavara) [reply src/mono/mono/utils/mono-threads-api.h:30]: That's probably better solution.

**Review by lambdageek — APPROVED** [review]

**Comment by lambdageek** [comment]
FYI @vitek-karas @ivanpovazan

**Comment by ivanpovazan** [comment]
@lambdageek could we use similar approach to improve what was called out here: https://github.com/dotnet/android/issues/8832#issuecomment-2133462990

**Pull request merged**